src/Stratis.Bitcoin.Features.Wallet/WalletSyncManager.cs

```diff
@@ -111,8 +111,13 @@ private void OnTransactionAdded(TransactionAddedToMemoryPool transactionAddedToM
 
         private void OnTransactionRemoved(TransactionRemovedFromMemoryPool transactionRemovedFromMempool)
         {
-            this.logger.LogDebug("Removing transaction '{0}' as it was removed from the mempool.", transactionRemovedFromMempool.RemovedTransaction.GetHash());
-            this.walletManager.RemoveUnconfirmedTransaction(transactionRemovedFromMempool.RemovedTransaction);
+            this.logger.LogDebug("Transaction '{0}' was removed from the mempool. RemovedForBlock={1}", 
+                transactionRemovedFromMempool.RemovedTransaction.GetHash(), transactionRemovedFromMempool.RemovedForBlock);
+
+            if (!transactionRemovedFromMempool.RemovedForBlock)
```

Comment:
Since we are synchronizing unconfirmed transactions in the wallet with unconfirmed transactions in the mempool it may be better not to omit information about the addition/removal of unconfirmed transactions.

Reply:
The main reason for doing this is when a transaction is included in a block, it is weird to remove it from the wallet in the meantime. 

There should be some continuity when the transaction is confirmed in a block. It's not a new entry, it's the same one. An unconfirmed -> confirmed and so we shouldn't remove it from our wallet in the meantime.

Reply:
There's also a potential race condition if we continue to remove the transaction from the wallet for a short time. If the user builds a transaction in that window, it will use transactions that may have _just_ been included in a block.